### PR TITLE
Contributing: setup instruction update, and more

### PR DESCRIPTION
**Pull request by chalin**
- Updated dev setup instructions. Also link to Gitpod.
- Replace "mirroring" section by current info.

Preview:

- https://github.com/chalin/opentelemetry.io/blob/chalin-dev-doc-updates-2021-11-17/CONTRIBUTING.md
- https://github.com/chalin/opentelemetry.io/blob/chalin-dev-doc-updates-2021-11-17/README.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,26 +15,20 @@ To add your project, please make a pull request. You'll need to create a data fi
...
-- [Morgan McLean](https://github.com/mtwo), Google
-- [Austin Parker](https://github.com/austinlparker), LightStep
-
-Learn more about roles in the [community repository](https://github.com/open-telemetry/community/blob/main/community-membership.md).
-
-Thanks to all the people who already contributed!
-
-<a href="https://github.com/open-telemetry/opentelemetry.io/graphs/contributors">
-  <img src="https://contributors-img.web.app/image?repo=open-telemetry/opentelemetry.io" />
-</a>
```
Comment (chalin) [inline README.md:33]: I removed the image, IMHO it is visually overpowering w/o adding much value. I kept the link to the contributors tab.

File: README.md [hunk 2]
```diff
@@ -14,26 +15,20 @@ To add your project, please make a pull request. You'll need to create a data fi
...
+Thanks to [all who have already
+contributed](https://github.com/open-telemetry/opentelemetry.io/graphs/contributors)!
 
 ## Documentation License
```
Comment (chalin) [inline README.md:27]: This sections needs some updating, but I'll address that later.

File: README.md [hunk 3]
```diff
@@ -14,26 +15,20 @@ To add your project, please make a pull request. You'll need to create a data fi
...
+- Learn more about roles in the [community
+  repository](https://github.com/open-telemetry/community/blob/main/community-membership.md).
 
-- [Shelby Spees](https://github.com/shelbyspees), Honeycomb
-
-Maintainers ([@open-telemetry/docs-maintainers](https://github.com/orgs/open-telemetry/teams/docs-maintainers))
-
-- [Steve Flanders](https://github.com/flands), Splunk
-- [Morgan McLean](https://github.com/mtwo), Google
-- [Austin Parker](https://github.com/austinlparker), LightStep
```
Comment (chalin) [inline README.md:25]: I removed the lists of names because they were outdated. The link to the group is good enough IMHO.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,54 +1,83 @@
...
-Netlify re-builds and re-deploys the site and stages a deploy preview for those
-changes.
-
-> Site administrators can access the admin interface
-> [here](https://app.netlify.com/sites/opentelemetry/overview).
-
-[localhost]: http://localhost:8888
+## Dev setup
+
+Follow these steps for local or [cloud-IDE][] (via [Gitpod.io][]) development:
```
Comment (chalin) [inline CONTRIBUTING.md:5]: The Gitpod IDE is amazing! I've easily been able to make site contributions using it (not as my main work environment, but it's great when working from elsewhere).

Comment (austinlparker) [reply CONTRIBUTING.md:5]: Do you know if the instructions work with Codespaces as well?

**Review by austinlparker — APPROVED** [review]

File: CONTRIBUTING.md (hunk 4 above)
Comment (chalin) [reply CONTRIBUTING.md:5]: No, I haven't tried it yet.

**Pull request merged**